Bug 1840446: Fix conditions table runtime error on operand details page

TheRealJon opened the pull request:
Avoid runtime error in `Conditions` component if the `conditions` prop is not an array.

TheRealJon commented:
/cherry-pick 4.4

TheRealJon commented:
/assign @spadgett

jcaianirh commented on frontend/public/components/conditions.tsx:
```diff
@@ -1,11 +1,11 @@
 import * as React from 'react';
-
+import * as _ from 'lodash';
```
not using lodash can remove this import statement.

spadgett commented on frontend/public/components/conditions.tsx:
```diff
@@ -1,11 +1,11 @@
 
-export const Conditions: React.SFC<ConditionsProps> = ({ conditions }) => {
-  const rows = (conditions || []).map((condition: K8sResourceCondition, i: number) => (
+export const Conditions: React.SFC<ConditionsProps> = ({ conditions = [] }) => {
+  const rows = conditions.map?.((condition: K8sResourceCondition, i: number) => (
```
This will cause a runtime error if we get `null` as a value instead of `undefined`.

```suggestion
export const Conditions: React.SFC<ConditionsProps> = ({ conditions }) => {
  const rows = conditions?.map?.((condition: K8sResourceCondition, i: number) => (
```

TheRealJon replied:
Fixed

spadgett commented:
> If you viewed the resource from Home -> Search, you would see the same error.

I guess it would no longer be a runtime error, but we'd incorrectly show the conditions table.

TheRealJon commented:
@spadgett @jcaianirh Addressed your comments and tweaked the solution a little bit. I figured there's no need to to a null check and an array check before rendering the conditions section on details pages. The array check covers both cases.
